File: backend/app/schemas/recommendation.py
from pydantic import BaseModel, Field


class CategoryScoreInput(BaseModel):
    category: str
    score: float = Field(ge=0, le=100)
    label: str


class RecommendedProduct(BaseModel):
    platform: str
    platform_label: str
    product_name: str
    brand: str
    image_url: str | None = None
    price: float | None = None
    reason: str
    match_score: float


class PlatformRecommendationResponse(BaseModel):
    worst_metric: str
    worst_metric_label: str
    worst_score: float
    products: list[RecommendedProduct]

File: frontend/src/api/recommendation.ts
import client from "./client";

export interface CategoryScoreInput {
  category: string;
  score: number;
  label: string;
}

export interface RecommendedProduct {
  platform: string;
  platform_label: string;
  product_name: string;
  brand: string;
  image_url: string | null;
  price: number | null;
  reason: string;
  match_score: number;
}

export interface PlatformRecommendationResponse {
  worst_metric: string;
  worst_metric_label: string;
  worst_score: number;
  products: RecommendedProduct[];
}

export async function getPlatformRecommendations(
  categories: CategoryScoreInput[],
): Promise<PlatformRecommendationResponse> {
  const { data } = await client.post<PlatformRecommendationResponse>(
    "/v1/recommendations/platform",
    categories,
  );
  return data;
}
